### DISCUSSION TITLE
How do I undo: git push origin my_branch --force

### DISCUSSION THREAD

**Discussion by berk-canc** [community/community#142108] [New to GitHub] [answered]
Here is what happened. I screwed up merge conflicts and pushed changes that are bad. I wanted to "revert" or "get rid of" that commit and go back to the previous commit.

I did this:

```
git reset --hard <commit_i_want>
git push origin <my_branch> --force
```

Now, all I have in my branch are the changes from <commit_i_want> only. All other commits before that commit are gone.

In other words, I wanted to remove commit4 only:

`commit1 --- commit2 --- commit3 --- commit3 --- commit4 (head)`

What I have now is commit3 only, everything else is before that is gone. How can I fix this? Thx!

**Comment by yolorabbit777** [comment, marked as answer]
Hi @berk-canc ,

Hope you are doing well.

To recover the missing commits, you can leverage Git reflog. Reflog records all your local changes, including those from before the git reset and force push. Here’s how you can restore your branch:

1. Check the Reflog to Find the Lost Commits: Run the following command to see the history of all actions on your branch (including before the reset):
`git reflog`
Then you will see output similar to this.
```
a1b2c3d HEAD@{0}: reset: moving to <commit_i_want>
e5f6g7h HEAD@{1}: commit: commit4
9a8b7c6 HEAD@{2}: commit: commit3
...
```

2. Reset to the Commit Before the Problem Occured
`git reset --hard 9a8b7c6`

3. Force-Push the Correct History
`git push origin <my_branch> --force`

I wish this will help you to solve the problem.

Best,
@yolorabbit

**Reply by berk-canc** [reply]
It worked, thank you so much!

**Comment by Aninooo** [comment]
1. Check the reflog: Run the following command to see the history of your HEAD:
   - git reflog
  
   Look for the commit hash of `commit4` or the state before your reset.

2. Reset to the correct commit: Once you find the commit hash (let's call it `<commit_hash>`), you can reset your branch back to that commit:
  - git reset --hard <commit_hash>

3. Push the changes: Since you've reset your branch, you'll need to force push again to update the remote:
  - git push origin <my_branch> --force
